### PR TITLE
[Term Entry] PyTorch function: tensor_split()

### LINKED ISSUE #5927: [Term Entry] PyTorch Tensor Operations: .tensor_split()
### Reason/inspiration (optional)

We would like a new entry on the `.tensor_split()` term under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by mwmartella**
### Description

- Added tensor-split term for PyTorch
- Closes #5927 

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+Title: 'tensor-split'
+Description: 'Takes a input for a tensor and splits it into multiple tensors.'
+Subjects:
+  - 'Data Science'
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:5]:
```suggestion
Subjects:
  - 'AI'
  - 'Data Science'
  - 'Machine Learning'
```
Added some more subjects that are supported by `tensor-split` term in `tensor-operations.`

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Tags:
+  - 'Data Structures'
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:7]:
```suggestion
  - 'Data Structures'
  - 'Deep Learning'
  - 'PyTorch'
  - 'Tensor'
```
Added some more tags for better understanding and relevance.

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/data-science'
+---
+
+In PyTorch, the **`.tensor_split()`** function splits a tensor into multiple sub-tensors. PyTorch will attempt to split the tensor evenly but when the tensor cannot be split evenly it will create as close to even sub-tensors as possible. The sub tensors can be also shaped with a dim (dimension) input.
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:13]:
```suggestion
In PyTorch, the **`.tensor_split()`** function splits a tensor into multiple sub-tensors. If the tensor cannot be split evenly, PyTorch distributes the elements as evenly as possible. The shape of the sub-tensors can also be controlled using the `dim` (dimension) parameter.
```
Reframed the sentence for more consistency and grammatical correctness.

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## Syntax
+
+```pseudo
+torch.tensor_split(input, indices_or_sections, dim=0)
+```
+
+- `input(Tensor)`: The tensor to be split.
+
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:22]:
```suggestion

```
Please remove this extra blank line over here.

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- `indices_or_sections(int)`: How many sub tensors to split into
+
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:24]:
```suggestion

```
Please remove this extra blank line over here.

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+## Syntax
+
+```pseudo
+torch.tensor_split(input, indices_or_sections, dim=0)
+```
+
+- `input(Tensor)`: The tensor to be split.
+
+- `indices_or_sections(int)`: How many sub tensors to split into
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:23]:
```suggestion
- `indices_or_sections(int)`: Number of sub-tensors to split the input tensor into, or the indices at which to split along the specified dimension.
```
This sentence seems grammatically incorrect or incomplete. So rephrase it for more clarity.

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+- `dim(int, optional)`: Change the dimensions of the split tensors, Default=0
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:25]:
```suggestion
- `dim(int, optional)`: The dimension along which to split the tensor. Default is `0`.
```

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+## Example
+
+The following example demonstrates the usage of the `.tensor_split()` function:
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:29]:
```suggestion
The following example demonstrates the use of the `.tensor_split()` function:
```

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+```py
+import torch
+
+x = torch.arange(10) #create a one dimensional tensor
+torch.tensor_split(x, 2) #split the tensor into 2 parts
+```
+```shell
+#the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each subtensor has 5 elements
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:38]:
```suggestion
#the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each subtensor has 5 elements
```
Instead of this comment: "#the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each sub-tensor has 5 elements," you can use the following sentence before the shell block: "The following code splits the tensor into 2 equal sub-tensors. Since 10 can be evenly divided by 2, each sub-tensor will contain 5 elements:"

File: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+```py
+import torch
+
+x = torch.arange(13) #create a one dimensional tensor
+torch.tensor_split(x, 6) #split the tensor into 6 parts
+```
+```shell
+#the above code produces the tensor in 6 almost equal parts. 13 is not divisable by 6 so PyTorch does this as close to even as possible the first sub tensor having 3 elements and the rest only 2.
+(tensor([0, 1, 2]), tensor([3, 4]), tensor([5, 6]), tensor([7, 8]), tensor([ 9, 10]), tensor([11, 12]))
+```
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md:51]:
```suggestion
```py
import torch

x = torch.arange(13) #create a one dimensional tensor
torch.tensor_split(x, 6) #split the tensor into 6 parts
```
```shell
#the above code produces the tensor in 6 almost equal parts. 13 is not divisable by 6 so PyTorch does this as close to even as possible the first sub tensor having 3 elements and the rest only 2.
(tensor([0, 1, 2]), tensor([3, 4]), tensor([5, 6]), tensor([7, 8]), tensor([ 9, 10]), tensor([11, 12]))
```
```
This section can be removed since an example has already been added earlier in this entry.

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey @mwmartella,
The entry is well-written! I’ve suggested a few changes—please make them as soon as possible. Also, please note that being on the main branch is discouraged. Please create a new separate branch for this entry. Thanks for contributing to the Codecademy docs! :) Feel free to reach out if you have any questions.

Thanks,
Savi
